ceph/exporter.go

```diff
@@ -39,22 +39,36 @@ type Exporter struct {
 	RbdMirror bool
 	Logger    *logrus.Logger
 	Version   *Version
+	cc        []prometheus.Collector
 }
 
 // NewExporter returns an initialized *Exporter
 // We can choose to enable a collector to extract stats out of by adding it to the list of collectors.
 func NewExporter(conn Conn, cluster string, config string, user string, rgwMode int, logger *logrus.Logger) *Exporter {
-	return &Exporter{
+	e := &Exporter{
 		Conn:    conn,
 		Cluster: cluster,
 		Config:  config,
 		User:    user,
 		RgwMode: rgwMode,
 		Logger:  logger,
 	}
+	err := e.setCephVersion()
+	if err != nil {
+		e.Logger.WithError(err).Error("failed to set rbd mirror")
```

Comment:
:spaghetti:

Reply:
yeah it's in 3 places now...

Reply:
But yes, it would be great to factor this out into a common method.